repositories/image_common.BUILD.bazel

```diff
@@ -0,0 +1,147 @@
+load(
+    "@com_github_mvukov_rules_ros2//ros2:cc_defs.bzl",
+    "ros2_cpp_library",
+)
+load(
+    "@com_github_mvukov_rules_ros2//ros2:interfaces.bzl",
+    "cpp_ros2_interface_library",
+)
+load(
+    "@com_github_mvukov_rules_ros2//ros2:plugin.bzl",
+    "ros2_plugin",
+)
+load("@rules_cc//cc:defs.bzl", "cc_library")
+
+ros2_cpp_library(
+    name = "camera_calibration_parsers",
+    srcs = glob([
+        "camera_calibration_parsers/src/parse.cpp",
+        "camera_calibration_parsers/src/parse_ini.cpp",
+        "camera_calibration_parsers/src/parse_yml.cpp",
+    ]),
+    hdrs = glob([
+        "camera_calibration_parsers/include/**/*.h",
+        "camera_calibration_parsers/include/**/*.hpp",
+    ]),
+    includes = ["camera_calibration_parsers/include"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@ros2_common_interfaces//:cpp_sensor_msgs",
+        "@ros2_rclcpp//:rclcpp",
+        "@ros2_rcpputils//:rcpputils",
+        "@yaml-cpp",
+    ],
+)
+
+ros2_cpp_library(
+    name = "image_transport",
```

Comment:
Didn't look into the original code thoroughly, but, I would expect that the plugin could be data dep of this target (that way the plugin also doesn't need the public visibility). Please check if this is possible.

Furthermore, I would expect a structure as following:
```Bazel

cc_library(
  name = "image_transport_common_hdrs",
  hdrs = <explicitly defined>,
  ...
)

ros2_cpp_library(
  name = "image_transport",
  deps = [":image_transport_common_hdrs"],
  data = [":raw_pub_sub],
  hdrs = [],  # explicitly defined such that deps are clear,
  # should not include common (as those come in via deps), raw_publisher, raw_subscriber hdrs.
  ...
)

cc_library(
  name = "raw_publisher",
  deps = [":image_transport_common_hdrs"],
  ...
)

cc_library(
  name = "raw_subscriber",
  deps = [":image_transport_common_hdrs"],
  ...
)

ros2_plugin(
  name = "raw_pub_sub_plugin",
  srcs = [...],
  deps = [":raw_publisher", "raw_subscriber"],
)

```

Reply:
On a second thought, image_transport should depend on raw_publisher and raw_subscriber as well -- but only on the interfaces, not on any code related to the concrete impl of plugins.

Reply:
I updated the code.